src/netaddress.h

```diff
@@ -272,6 +370,65 @@ class CNetAddr
 
             UnserializeV1Array(serialized);
         }
+
+        /**
+         * Unserialize from a ADDRv2 / BIP155 format.
+         */
+        template <typename Stream>
+        void UnserializeV2Stream(Stream& s)
+        {
+            uint8_t bip155_net;
+            s >> bip155_net;
+
+            size_t address_size;
+            s >> COMPACTSIZE(address_size);
+
+            if (address_size > MAX_ADDRV2_SIZE) {
+                throw std::ios_base::failure(strprintf(
+                    "Address too long: %u > %u", address_size, MAX_ADDRV2_SIZE));
+            }
+
+            scopeId = 0;
```

Comment:
I don't see the `scopeId` variable being used in this block, could it be removed? Or is it required to be set here for something else down the line?

Reply:
It's a member variable of `CNetAddr`, so yes, it is used later.